scipy/_lib/_array_api.py

```diff
@@ -276,6 +279,9 @@ def _strict_check(actual, desired, xp, *,
         assert actual.dtype == desired.dtype, _msg
 
     if check_shape:
+        if is_dask(xp):
+            actual.compute_chunk_sizes()
+            desired.compute_chunk_sizes()
```

Comment:
@jakirkham does your comment about assignment also apply here?

Reply:
Good question. Not in this case

Basically `compute_chunk_sizes` is trying to replace unknown chunk sizes with known chunk sizes. This works in-place ([ref]( https://docs.dask.org/en/stable/generated/dask.array.Array.compute_chunk_sizes.html#dask.array.Array.compute_chunk_sizes ))

Here is a quick example of how unknown chunks are handled by `compute_chunk_sizes`

Code:

```python
import dask.array as da

a = da.arange(9, chunks=3)

a2, = da.where(a % 3 == 0)
print(a2)

a2.compute_chunk_sizes()
print(a2)
```

Output:

```
dask.array<getitem, shape=(nan,), dtype=int64, chunksize=(nan,), chunktype=numpy.ndarray>
dask.array<getitem, shape=(3,), dtype=int64, chunksize=(1,), chunktype=numpy.ndarray>
```